pkg/agent/multicast/mcast_discovery.go

```diff
@@ -97,14 +100,48 @@ func (s *IGMPSnooper) queryIGMP(group net.IP, versions []uint8) error {
 		if err != nil {
 			return err
 		}
-		if err := s.ofClient.SendIGMPQueryPacketOut(igmpQueryDstMac, mcastAllHosts, openflow13.P_NORMAL, igmp); err != nil {
+		outPort := uint32(0)
+		if err := s.ofClient.SendIGMPQueryPacketOut(igmpQueryDstMac, types.McastAllHosts, outPort, igmp); err != nil {
 			return err
 		}
-		klog.V(2).InfoS("Sent packetOut for IGMP query", "group", group.String(), "version", version)
+		klog.V(2).InfoS("Sent packetOut for IGMP query", "group", group.String(), "version", version, "outPort", outPort)
 	}
 	return nil
 }
 
+func (s *IGMPSnooper) validate(event *mcastGroupEvent) (bool, error) {
+	if !s.anpEnabled {
```

Comment:
Can we use `s.Validator == nil` as the condition? Then we don't need to give the parameter `anpEnabled`. And in the constructor for Multicast Controller, if ANP is enabled, networkPolicyController is given, otherwise, "nil" is given.

Reply:
done